### PR TITLE
JENKINS-45099 special JSON characters are not consistently quoted

### PR DESCRIPTION
**Pull request by i386**
Special JSON characters such as `[` and `]` were not consistently quoted when writing to the output stream, resulting in invalid JSON.

This change uses `commons-lang` v3's `org.apache.commons.lang3.StringEscapeUtils#escapeJson` which ensures that all special JSON characters and control characters are encoded correctly. 

PTAL @reviewbybees @jglick @vivek

### CODE REVIEW

**Comment by michaelneale** [comment]
LGTM

**Comment by stephenc** [comment]
Would be great if this was as two commits to show the failing test on CI first and then show how this change fixes... because I'm not seeing the encoding issue in the test case

I would also expect tests covering Unicode escaping that escapeJson adds to the mix

File: core/pom.xml [hunk 1]
```diff
@@ -47,6 +47,11 @@
       <version>1.7.0</version>
     </dependency>
     <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-lang3</artifactId>
+      <version>3.5</version>
```
Comment (oleg-nenashev) [inline core/pom.xml]: Jenkins widely uses 3.x: https://github.com/search?l=Maven+POM&p=2&q=org%3Ajenkinsci+%22commons-lang3%22&type=Code . The dependency should be fine

**Comment by i386** [comment]
@stephenc I added the extra tests you asked for.

File: core/pom.xml (hunk 1 above)
Comment (i386) [inline core/pom.xml]: could always shade it in to be safe.

File: core/src/test/java/org/kohsuke/stapler/export/JSONDataWriterTest.java [hunk 2]
```diff
@@ -18,6 +18,31 @@
...
+        public String getValue() {
+            return "\"'/[]{}\\";
+        }
+    }
+
+    @ExportedBean
+    public static class JsonCharacters2 {
+        @Exported
+        public String getValue() {
+            return "\"foo\" isn't \"bar\". specials: \b\r\n\f\t\\/";
```
Comment (stephenc) [inline core/src/test/java/org/kohsuke/stapler/export/JSONDataWriterTest.java]: I'm not seeing any tests covering the change this brings to escaping Unicode characters (specifically the method you are using will use the `\u0080` for any codepoint above 0x80... not strictly necessary for JSON as that should be in a UTF encoding always

**Comment by michaelneale** [comment]
ping @i386

**Comment by i386** [comment]
@michaelneale I don't think this is a quick fix and probably a suckhole for my time.

**Comment by i386** [comment]
@stephenc thanks for the review feedback. Ill find someone else to look at this.

**Comment by michaelneale** [comment]
@i386 right - does https://github.com/jenkinsci/blueocean-plugin/pull/1281 need to be undone on the blue side of things now or is that different? 

This seems to be a deep yak hole
